opsdroid/configuration/validation.py

```diff
@@ -0,0 +1,33 @@
+"""Schema validation for configuration.yaml."""
+
+from voluptuous import Schema, ALLOW_EXTRA
+
+
+schema = {
+    "logging": {"level": bool, "console": bool},
+    "welcome-message": bool,
+    "connectors": [{"name": str, "token": str, "access-token": str}],
```

Comment:
Not all connectors have a `token` or `access-token` here, is that ok?

Reply:
Yeah it's alright because the lib will check if the key exists, if it does it will check for type if not it will skip which is great when it comes to some connectors. 
This schema is just a test one I still need to test it further since I'm having issues with `parsers` and `databases` because when they are set it will be a list, when they aren't set it will be just None so I need to check how to deal with it